metadata-ingestion/src/datahub/ingestion/source/grafana/grafana_config.py

```diff
@@ -0,0 +1,109 @@
+    )
+    database: Optional[str] = Field(default=None, description="Default database name")
+    database_schema: Optional[str] = Field(
+        default=None, description="Default schema name"
+    )
+
+
+class GrafanaSourceConfig(
+    DatasetLineageProviderConfigBase,
+    StatefulIngestionConfigBase,
+    PlatformInstanceConfigMixin,
+    EnvConfigMixin,
+):
+    """Configuration for Grafana source"""
+
+    platform: str = Field(default="grafana", hidden_from_docs=True)
+    url: str = Field(
+        description="Grafana URL in the format http://your-grafana-instance with no trailing slash"
+    )
+    service_account_token: SecretStr = Field(
+        description="Service account token for Grafana"
+    )
+    verify_ssl: bool = Field(
+        default=True,
+        description="Whether to verify SSL certificates when connecting to Grafana",
+    )
+
+    # API pagination configuration
+    page_size: int = Field(
+        default=100,
+        description="Number of items to fetch per API call when paginating through folders and dashboards",
+    )
+
+    # Extraction mode configuration
+    basic_mode: bool = Field(
+        default=False,
+        description="Enable basic extraction mode for users with limited permissions. "
+        "In basic mode, only dashboard metadata is extracted without detailed panel information, "
+        "lineage, or folder hierarchy. This requires only basic dashboard read permissions.",
+    )
+
+    # Content filtering
+    dashboard_pattern: AllowDenyPattern = Field(
+        default=AllowDenyPattern.allow_all(),
+        description="Regex pattern to filter dashboards for ingestion",
+    )
+    folder_pattern: AllowDenyPattern = Field(
+        default=AllowDenyPattern.allow_all(),
+        description="Regex pattern to filter folders for ingestion",
+    )
+
+    # Feature toggles
+    ingest_tags: bool = Field(
+        default=True, description="Whether to ingest dashboard and chart tags"
+    )
+    ingest_owners: bool = Field(
+        default=True, description="Whether to ingest dashboard ownership information"
+    )
+
+    # Lineage configuration
+    extract_lineage: bool = Field(
+        default=True,
+        description="Whether to extract lineage between charts and data sources. "
+        "When enabled, the source will parse SQL queries and datasource configurations "
+        "to build lineage relationships.",
+    )
+    include_column_lineage: bool = Field(
+        default=True,
+        description="Whether to extract column-level lineage from SQL queries. "
+        "Only applicable when extract_lineage is enabled.",
+    )
+
+    # Platform connection mappings
+    connection_to_platform_map: Dict[str, PlatformConnectionConfig] = Field(
+        default_factory=dict,
+        description="Map of Grafana datasource types/UIDs to platform connection configs for lineage extraction",
+    )
+    stateful_ingestion: Optional[StatefulStaleMetadataRemovalConfig] = None
```

Comment:
is this needed considering the config already inherits from `StatefulIngestionConfigBase`?